MOM-2024-02-14 (#124)
camaraproject/IdentityAndConsentManagement: documentation/MeetingMinutes/2024/MOM-2024-02-14.md

# Identity and Consent Management meeting

#### *February 14, 2024*

<br>

> Preliminary Minutes of Meeting. To be updated during and after the meeting.

Reviewer: Next time, you might want to remove this "preliminary" comment when the final meeting minutes are ready, right? (It is not important anyway)

| Gapask    | Rajesh Murthy |
| GSMA      | Mark Cornall, Toyeeb Rehman, Tom van Pelt |
| KPN       | [Huub Appelboom](https://github.com/HuubAppelboom) |
| Nokia     | [Tanja De Groot](https://wiki.camaraproject.org/display/~TanjaDeGroot), [Gaurav Agarwal](https://wiki.camaraproject.org/display/~gaurav2192) |
| OIDF      | [Bjorn Hjelm (OIDF)](https://github.com/bhjelm), [Joseph Heenan](https://github.com/jogu)  |
| Shabodi   | Kevin Howe-Patterson |
| Singtel   | [Foo Ming Hui](https://wiki.camaraproject.org/display/~mhfoo) |
| Spry Fox Networks | [Ramesh Shanmugasundaram](https://wiki.camaraproject.org/display/~sfnuser), [Parichaya Shrivastava](https://wiki.camaraproject.org/display/~lfsfn) |
| T-Mobile PL | [Dawid Wroblewski](https://github.com/DT-DawidWroblewski), [Artych, Rafał](https://wiki.camaraproject.org/display/~rart) |
| T-Mobile US | [Karabulut, Murat](https://wiki.camaraproject.org/display/~gmuratk) |
| Telefónica | [Jesús Peña García-Oliva](https://github.com/jpengar), [Diego Gonzalez Martínez](https://github.com/diegogonmar), Guido García,<br> Juan Fabio García, Pedro Ballesteros, David Vallejo,<br> Juan Antonio Hernando, Diego Yonadi |

> | Vodafone  | Eric Murray, Sönke Peters, [Sachin Kumar](https://wiki.camaraproject.org/display/~sachinvodafone)|

Reviewer 2: I didn't attend this meeting

Author: thanks, corrected

## Discussion the Way Forward regarding the Profile

Jesús recapitulated the last meeting where PR #113 was discussed and that consensus was achieved.
Axel and Shilpa did not agree with consensus beeing achieved - see previous section.
Axel said that he created the issue https://github.com/camaraproject/IdentityAndConsentManagement/issues/122 and kindly asked WG members to comment on the issue.
Axel also created PR https://github.com/camaraproject/IdentityAndConsentManagement/pull/121 and asked people to review it.

After some discussion we agreed that WG members comment on the issue and review the PR seeking resolution and a decision whether to use the DT proposal or the TEF proposal as our WG's policy document.
Elizabeth suggested that if discussions on GitHub get too long, we would call for a side meeting, anounce it on the mailing this and seek consensus.

Axel agreed with this proposal and no objections where heard.

Axel asked Bjorn Hjelm for his/OIDF's opinion. Bjorn said that OIDF is here to help.

We agreed that if neither the commenting on the issue nor the review of the PR lead to consensus, the WG seek a vote at TSC.

> ## Discussion about [offline-access PR](https://github.com/camaraproject/IdentityAndConsentManagement/pull/123)

Reviewer 3: ```suggestion
## Discussion about [offline-access and Refresh Token PR](https://github.com/camaraproject/IdentityAndConsentManagement/pull/123)
```

# Identity and Consent Management meeting

#### *February 14, 2024*

<br>

Preliminary Minutes of Meeting. To be updated during and after the meeting.

> ## Attendees

Reviewer 4: Not related to the cosmetic suggestion, just about the format: 

Using the https://wiki.camaraproject.org/display/~xxx links is the recommended way of LF for listing the meeting participants if minutes are done within the Confluence wiki (which is the target for CAMARA). You only need to create a LF account to have access to the information (which every participant within CAMARA should have eventually). 

But I agree that the format might be not used as long the meeting minutes are in GitHub.

> ## Attendees

Reviewer: I have access to these links, but other people don't. But like I said, I'm fine with whatever Axel decides. No problem at all.

> ## Attendees

Author: I added a table of attendees, and added the Camara wiki links. 
I 'll add the githup links in future moms.

@jpengar please review the TEF participants

[Gaurav Agarwal](https://wiki.camaraproject.org/display/~gaurav2192)

[Herbert Damker](https://wiki.camaraproject.org/display/~hdamker)

[Jesús Peña García-Oliva](https://github.com/jpengar)

[Diego Gonzalez Martínez](https://github.com/diegogonmar)

Sönke Peters, Vodafone

[Bjorn Hjelm (OIDF)](https://github.com/bhjelm) 

[Dawid Wroblewski](https://github.com/DT-DawidWroblewski)

[Huub Appelboom](https://github.com/HuubAppelboom)

[Tanja De Groot](https://wiki.camaraproject.org/display/~TanjaDeGroot)

[Karabulut, Murat](https://wiki.camaraproject.org/display/~gmuratk)

Toyeeb Rehman (GSMA)

Rajesh Murthy 

[Padgaonkar, Shilpa](https://wiki.camaraproject.org/display/~shilpa.padgaonkar) 

[Joseph Heenan](https://github.com/jogu)

[Artych, Rafał](https://wiki.camaraproject.org/display/~rart)

[Surajj Jaggernath](https://wiki.camaraproject.org/display/~surajjj)

[Parichaya Shrivastava](https://wiki.camaraproject.org/display/~lfsfn)

[Elisabeth Mueller](https://wiki.camaraproject.org/display/~elisabethmueller)

[Foo Ming Hui](https://wiki.camaraproject.org/display/~mhfoo)

[Sachin Kumar](https://wiki.camaraproject.org/display/~sachinvodafone)

DAVID VALLEJO NAVARRO

> [Diego González Martínez](https://wiki.camaraproject.org/display/~diegogonmar)

Author: Moved the two entries together and continue work on it tomorrow.

<br>

## Agenda

1. Recent updates & recap
2. Issues and PRs. Priority discussions (most active issues and/or dependencies for release v0.1):
    - **Camara profile of OIDF and IETF standards**
        - Please comment on https://github.com/camaraproject/IdentityAndConsentManagement/issues/122 .
    - **offline access** #123 https://github.com/camaraproject/IdentityAndConsentManagement/pull/123
    - **Discussion on Issue #100 "Claims to be used in oAuth Access tokens".**
    - **Discussion on Issue #108 "Question to documentation update for security schemes".**
    - **Discussion on Issue #104 "Alignment of Client Credential authentication / authorization using API Gateway Pattern".**

3. Action Points
4. AoB

<br>

## Meeting Technicallities

Jesús offered that ownership of the meeting invite could be transferred to the new chair. Axel said that for now this can stay as it is.

## Discussion on Consensus in Camara Working Groups

Recapitulation of the previous discussion on this topic: Jesús and Diego said there was consensus achieved during a WG call that we continue working on https://github.com/camaraproject/IdentityAndConsentManagement/pull/113

> Herbert said that lazy-consensus is the Camara way. Shilpa said no consensus was achieved. 

Reviewer 4: Why we need to document this discussion about the previous meeting at all in the minutes - it isn't an outcome which brings us forward. Objections were raised (if you want: after the meeting, based on the minutes) so the discussion had to be reopened (there wasn't a formal vote which would require a supermajority vote to reopen the decision).

Normally decision in a sub project meeting will be confirmed by conclusive actions within GitHub (e.g. approvals of a discussed PR). Here we have now the case that there is the discussion reopened as a GItHub issue (#122) where everyone can raise their voice and which hopefully will lead to a consensus, or if not to a vote in TSC.

> Herbert said that lazy-consensus is the Camara way. Shilpa said no consensus was achieved. 

Reviewer: I personally didn't need to document this discussion about the previous meeting in the minutes, but since it was included by Axel, I just wanted to include Telefónica's position and clarifications for the record. Because it was missing.

> Axel proposed that consensus should be achieved on Github because not everybody can attend meetings. Diego said that is not the Camara way.

Reviewer 4: See above https://github.com/camaraproject/IdentityAndConsentManagement/pull/124/files#r1490980472

## Discussion the Way Forward regarding the Profile

Jesús recapitulated the last meeting where PR #113 was discussed and that consensus was achieved.
Axel and Shilpa did not agree with consensus beeing achieved - see previous section.
Axel said that he created the issue https://github.com/camaraproject/IdentityAndConsentManagement/issues/122 and kindly asked WG members to comment on the issue.
Axel also created PR https://github.com/camaraproject/IdentityAndConsentManagement/pull/121 and asked people to review it.

After some discussion we agreed that WG members comment on the issue and review the PR seeking resolution and a decision whether to use the DT proposal or the TEF proposal as our WG's policy document.
Elizabeth suggested that if discussions on GitHub get too long, we would call for a side meeting, anounce it on the mailing this and seek consensus.

Axel agreed with this proposal and no objections where heard.

Axel asked Bjorn Hjelm for his/OIDF's opinion. Bjorn said that OIDF is here to help.

We agreed that if neither the commenting on the issue nor the review of the PR lead to consensus, the WG seek a vote at TSC.

## Discussion about offline-access

> Jesús said that this PR "just" moves documentation from some place (GSMA) to another and that this WG agreed on moving documentation to Camara as the "place of truth" for both GSMA and Camara.

Reviewer: ```suggestion
Jesús said that as previously discussed and agreed in the WG, and as per CAMARA TSC (Technical Steering Committee) decision, all documentation related to the NBI CAMARA API interface will only be placed in CAMARA. CAMARA will be the source of truth for northbound interface. GSMA Opengateway will simply adopt what is defined in CAMARA. And Jesús suggests that we first focus on moving the necessary documentation to the CAMARA Identity & Consent WG repository as agreed. And then, if the WG participants want to revisit the content of the documentation, we can open a new issue for that, no problem. This would allow us to wrap up a long discussion, close issue https://github.com/camaraproject/IdentityAndConsentManagement/issues/82, and move on to the next needed discussion without blocking this track by creating new dependencies between issues
```

> And example was given that some Commonalities document was moved to ICM and this is similar. Shilpa objected that moving a document from Camara to Camara is different.

Reviewer: ```suggestion
And example was given that some Commonalities document was moved to ICM and this is similar. Shilpa objected that moving a document from Camara to Camara is different.
Jesus disagrees because the AuthN-Auth Z document also contained information, for example, about authorization flows that were not agreed upon in the I&CM working group and other features that were not even discussed. And in fact, that document was reworked after being moved and even removed for the v0.2.0 release. Same process can be followed now.
```

Reviewer: Let's leave it at that, but it looks like you're suggesting that the GSMA Opengateway technical stream is not an open process. I don't think the GSMA would agree. And, AFAIK, many of the companies represented in this WG were also in the GSMA technical stream. But I agree, we are both right.

## Discussion on Consensus in Camara Working Groups

Recapitulation of the previous discussion on this topic: Jesús and Diego said there was consensus achieved during a WG call that we continue working on https://github.com/camaraproject/IdentityAndConsentManagement/pull/113
Herbert said that lazy-consensus is the Camara way. Shilpa said no consensus was achieved. 
Jesús disagrees and clarifies from his view of what happened in the previous call that the WG participants were asked, considering the different opinions, if we could agree as a way forward to continue with the "self-contained" contribution (PR #113) with compromise to properly note any changes related to the reused standard. And there was support from Ericsson, GSMA, and no objection from others on the call. DT of course preferred the delta approach, but Axel mentioned that he wouldn't disagree with the WG as long as the changes to the standard are properly marked in the document.

> Axel proposed that consensus should be achieved on Github because not everybody can attend meetings. Diego said that is not the Camara way and all CAMARA subprojects actually make decisions during the calls, and active WG participants are well aware of this.

Reviewer 3: Can we make clearer that proposal to not to allow decisions to be taken in meetings was dismissed? Otherwise it seems that there is an open discussion on this, which I think it is not.
This is compatible with following part of the minutes commenting about enhancing decision-making or with communication about decision to be taken in advance.

Author: added that my proposal was dismissed and that lazy-consensus it is in Camara WGs

## Discussion the Way Forward regarding the Profile

Jesús recapitulated the last meeting where PR #113 was discussed and that consensus was achieved.
Axel and Shilpa did not agree with consensus beeing achieved - see previous section.
Axel said that he created the issue https://github.com/camaraproject/IdentityAndConsentManagement/issues/122 and kindly asked WG members to comment on the issue.
Axel also created PR https://github.com/camaraproject/IdentityAndConsentManagement/pull/121 and asked people to review it.

After some discussion we agreed that WG members comment on the issue and review the PR seeking resolution and a decision whether to use the DT proposal or the TEF proposal as our WG's policy document.
Elizabeth suggested that if discussions on GitHub get too long, we would call for a side meeting, anounce it on the mailing this and seek consensus.

Axel agreed with this proposal and no objections where heard.

Axel asked Bjorn Hjelm for his/OIDF's opinion. Bjorn said that OIDF is here to help.

We agreed that if neither the commenting on the issue nor the review of the PR lead to consensus, the WG seek a vote at TSC.

> ## Discussion about offline-access

Reviewer 3: ```suggestion
## Discussion about offline-access and Refresh Token
```

Author: I added the link to the PR to the title

Reviewer 3: If you mention the PR instead, then mention Refresh Token as is in the name of the PR, which actually matches my suggestion that this discussion about offline-access and Refresh Token

Reviewer 3: @AxelNennker this one is pending, not sure if you saw it

And example was given that some Commonalities document was moved to ICM and this is similar. Shilpa objected that moving a document from Camara to Camara is different.
Jesus disagrees because the AuthN-Auth Z document also contained information, for example, about authorization flows that were not agreed upon in the I&CM working group and other features that were not even discussed. And in fact, that document was reworked after being moved and even removed for the v0.2.0 release. Same process can be followed now.
Axel asked meeting attendents to describe the Camara use case for offline-access and comment on the PR why Camara needs offline-access in ICM.

> Diego pointed out that ICM already has a document that talks about refresh tokens.

Reviewer 3: ```suggestion
Diego pointed out that ICM already has a document that indicates that Refresh Tokens are within covered scenarios, and the document is included in release v0.1.0 thus support for Refresh Token is already included. Diego indicates which document it is: https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md?plain=1#L240.
Also indicates that upon former agreement, the only thing being done is moving info to CAMARA for v0.2, but not changing scope. And that discussion for v0.3 may happen anyway
```

Axel replyied that OIDC offline-access is only offline-access to the user-info endpoint and that refresh tokens are a different thing.

> Diego said that he personally is not for or against this PR but that this PR just follows the "move documentation to Camara"-plan.

Reviewer 3: ```suggestion
Diego said that he personally is not for or against this PR but that this PR just follows the agreed way forward of moving documentation to Camara
```

We agreed that reviews should be written.

## Closing Remarks

Axel kindly asked participants to comment on issues and to review PRs and most importantly contribute text changes to PR.
WGs thrive only if members become participants. So, again please participate. Please comment and review.

<br>

## AoB

> - Next call schedule: February 31, 2024.

Reviewer 3: ```suggestion
- Next call schedule: February 28, 2024.
```
